### PR TITLE
[AWS] Add "Data Granularity" parameter for metrics data streams

### LINKED ISSUE #4246: [AWS] Add a new "Data Granularity" advanced parameter for metrics-related integrations
# Summary

Introduce a new parameter that customers can use to define the granularity of the datapoints returned by GetMetricData API call. 

# Task

Add `data_granularity` parameter (metricbeat's PR https://github.com/elastic/beats/issues/33133) to all datastreams relying on the `aws/cloudwatch` input. This parameter should be shown as “advanced” and not among the main ones (e.g period and region) and should be set by default to be equal to the collection period, if not defined by the customer. The current parameter “period” should be renamed to something like “collection period” to avoid confusion between this and the new granularity-related parameter.

## List of datastreams

- [ ] billing
- [ ] cloudwatch_metrics
- [ ] dynamodb
- [ ] ebs
- [ ] ec2_metrics
- [ ] ecs_metrics
- [ ] elb_metrics
- [ ] firewall_metrics
- [ ] kinesis
- [ ] lambda
- [ ] natgateway
- [ ] rds
- [ ] redshift
- [ ] s3_daily_storage
- [ ] s3_request
- [ ] s3_storage_lens
- [ ] sns
- [ ] sqs
- [ ] transitgateway
- [ ] usage
- [ ] vpn

### PR DESCRIPTION
**Pull request by gpop63**
## What does this PR do?

Adds `data_granularity` parameter for metrics data streams and renames `period` parameter title to `Collection Period`.

## Checklist

- [ ] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [ ] I have verified that all data streams collect metrics or logs.
- [ ] I have added an entry to my package's `changelog.yml` file.
- [ ] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Author's Checklist

- [ ]

## Related issues

- Closes #4246

### CODE REVIEW

File: packages/aws/changelog.yml [hunk 1]
```diff
@@ -1,4 +1,9 @@
 # newer versions go on top
+- version: "1.24.6"
```
Comment (kaiyan-sheng) [inline packages/aws/changelog.yml:2]: Let's bump it to 1.25.0 for a new feature :) Or even 2.0.0?

File: packages/aws/data_stream/billing/manifest.yml [hunk 2]
```diff
@@ -5,11 +5,18 @@ streams:
     vars:
       - name: period
         type: text
-        title: Period
+        title: Collection Period
         multi: false
         required: true
         show_user: true
         default: 12h
```
Comment (kaiyan-sheng) [inline packages/aws/data_stream/billing/manifest.yml:12]: I think this is a typo, should be 24h instead.

File: packages/aws/data_stream/billing/manifest.yml [hunk 3]
```diff
@@ -5,11 +5,18 @@ streams:
...
+      - name: data_granularity
+        type: text
+        title: Data Granularity
+        multi: false
+        required: false
+        show_user: false
+        default: 12h
```
Comment (kaiyan-sheng) [inline packages/aws/data_stream/billing/manifest.yml:19]: Should be 24h instead here too

File: packages/aws/manifest.yml [hunk 4]
```diff
@@ -1,7 +1,7 @@
 format_version: 1.0.0
 name: aws
 title: AWS
-version: 1.24.5
+version: 1.24.6
```
Comment (kaiyan-sheng) [inline packages/aws/manifest.yml:4]: This new parameter is added into Metricbeat 8.6.0 so we need to change Kibana version dependency here as well.

Comment (kaiyan-sheng) [reply packages/aws/manifest.yml:4]: In line 13

**Comment by kaiyan-sheng** [comment]
CI failure should be fixed by https://github.com/elastic/integrations/pull/4522

**Comment by girodav** [comment]
CI failures probably related to https://github.com/elastic/fleet-server/issues/2083

**Review by girodav — APPROVED** [review]

**Review by endorama — APPROVED** [review]

**Review by endorama — APPROVED** [review]

**Pull request merged**